CentralComputing/Configurator.cpp

```diff
@@ -4,33 +4,32 @@ using namespace std;
 
 Configurator ConfiguratorManager::config;
 
-int Configurator::openVarFile(string fileName) {
-    inFile.open(fileName);
-    if(!inFile) {
-        return 0;
-    }
-    loadValues();
-    return 1;
+bool Configurator::openConfigFile(const string& fileName) {
+  inFile.open(fileName);
+  if(!inFile) {
+    return false;
+  }
+  loadValues();
+  inFile.close();
+  return true;
 }
 
 void Configurator::loadValues() {
-    inFile.clear();
-    inFile.seekg(0, ios::beg);
-    string varName;
-    double val;
-    int count = 0;
-    while(inFile >> varName){
-	inFile >> val;
-	cout << varName << " " << val << " " << count<< endl;
-	count++;
-	mapVals.insert(pair<string, double> (varName, val));
-    }
+  inFile.clear();
+  inFile.seekg(0, ios::beg);
+  string varName;
+  double val;
+  while(inFile >> varName){
+    inFile >> val;
+    mapVals.insert(pair<string, double> (varName, val));
+  }
 }
 
-double Configurator::getValue(string varName) {
-    if (mapVals.find(varName) == mapVals.end()) {
-	return 0.0;
-    }
-    return mapVals.find(varName)->second;
+bool Configurator::getValue(const string& varName, double& value) {
```

Comment:
I changed the signature to `const string&`. `const` because we don't change the value of the string, and `&` which in C++ is a reference. Instead of copying the input string into this function (which is what would normally happen), this allows us to 'reference' the string without copying it. Copying large strings/ objects is expensive, and while the strings in this case are small, it is still good practice to use references `&`